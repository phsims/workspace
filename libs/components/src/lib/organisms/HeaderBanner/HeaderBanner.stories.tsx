import type { Meta, StoryObj } from '@storybook/react';
import { HeaderBanner, HeaderBannerProps } from './HeaderBanner';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<typeof HeaderBanner> = {
  component: HeaderBanner,
  title: '@organisms/HeaderBanner',
};
export default meta;
type Story = StoryObj<typeof HeaderBanner>;

export const Primary = {
  args: {
    background: 'grey.light',
    gridSplit: [9, 3],
    title: 'Find Your Favorite Recipes with Ease',
    intro:
      "Explore a world of culinary delights with our extensive collection of recipes. Whether you're looking for quick weeknight dinners, healthy meal options, or indulgent treats, our recipe cards offer something for everyone. Browse, save, and start cooking today!",
    image: {
      src: '/soup.svg',
      height: 526,
      width: 526,
      alt: 'Find Your Favorite Recipes',
    },
  } as HeaderBannerProps,
};

export const Heading: Story = {
  args: { ...Primary.args },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Find Your Favorite Recipes with Ease/gi)).toBeInTheDocument();
  },
};
