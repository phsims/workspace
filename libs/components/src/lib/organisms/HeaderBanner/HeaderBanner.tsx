import Image from 'next/image';
import { Container, Grid, Typography } from '@mui/material';
import Banner from '../../atoms/Banner/Banner';

export interface HeaderBannerProps {
  background: string;
  gridSplit: [3, 9] | [6, 6] | [9, 3];
  title: string;
  intro: string;
  image: {
    src: string;
    height: number;
    width: number;
    alt: string;
    imgTitle?: string;
  };
}

export function HeaderBanner({
  background,
  gridSplit = [6, 6],
  title,
  intro,
  image,
}: HeaderBannerProps) {
  const { src, height, width, alt, imgTitle = alt } = image;

  return (
    <Banner background={background}>
      <Container>
        <Grid container spacing={{ xs: 2, md: 3 }}>
          <Grid item md={gridSplit[0]} xs={1}>
            <Typography variant="h1" sx={{ mb: 3 }}>
              {title}
            </Typography>
            <Typography variant="body1"> {intro}</Typography>
          </Grid>
          <Grid item md={gridSplit[1]} xs={1}>
            <Image
              src={src}
              alt={alt}
              title={imgTitle}
              width={width}
              height={height}
              style={{
                width: '100%',
                height: 'auto',
              }}
            />
          </Grid>
        </Grid>
      </Container>
    </Banner>
  );
}

export default HeaderBanner;
