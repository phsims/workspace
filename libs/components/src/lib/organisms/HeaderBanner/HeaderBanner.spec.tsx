import { render } from '@testing-library/react';

import HeaderBanner, { HeaderBannerProps } from './HeaderBanner';

const mockProps: HeaderBannerProps = {
  gridSplit: [9, 3],
  title: 'Find Your Favorite Recipes with Ease',
  intro:
    "Explore a world of culinary delights with our extensive collection of recipes. Whether you're looking for quick weeknight dinners, healthy meal options, or indulgent treats, our recipe cards offer something for everyone. Browse, save, and start cooking today!",
  image: {
    src: '/soup.svg',
    height: 526,
    width: 526,
    alt: 'Find Your Favorite Recipes',
  },
};
describe('HeaderBanner', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<HeaderBanner {...mockProps} />);
    expect(baseElement).toBeTruthy();
  });
  it('should match snapshot', () => {
    const { asFragment } = render(<HeaderBanner {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  } );

  it('should render the correct title', () => {
    const { getByText } = render(<HeaderBanner {...mockProps} />);
    expect(getByText(mockProps.title)).toBeInTheDocument();
  });   
  it('should render the correct intro', () => {
    const { getByText } = render(<HeaderBanner {...mockProps} />);
    expect(getByText(mockProps.intro)).toBeInTheDocument();
  });
  it('should render the correct image', () => {
    const { getByAltText } = render(<HeaderBanner {...mockProps} />);
    expect(getByAltText(mockProps.image.alt)).toBeInTheDocument();
  });
      
});
