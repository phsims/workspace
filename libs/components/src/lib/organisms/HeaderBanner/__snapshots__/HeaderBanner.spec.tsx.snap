// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBanner should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1b2j6aq"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 css-1xgjpwf-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-md-9 css-hp6kmk-MuiGrid-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-yz1xox-MuiTypography-root"
          >
            Find Your Favorite Recipes with Ease
          </h1>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
             Explore a world of culinary delights with our extensive collection of recipes. Whether you're looking for quick weeknight dinners, healthy meal options, or indulgent treats, our recipe cards offer something for everyone. Browse, save, and start cooking today!
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-md-3 css-p80g4r-MuiGrid-root"
        >
          <img
            alt="Find Your Favorite Recipes"
            data-nimg="1"
            decoding="async"
            height="526"
            loading="lazy"
            src="/soup.svg"
            style="color: transparent; width: 100%; height: auto;"
            title="Find Your Favorite Recipes"
            width="526"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
